apps/go-mod-validator/test/github.test.ts

```diff
@@ -101,47 +115,62 @@ describe("isGoModReferencingDefaultBranch", () => {
     };
 
     it("should check if the commit is present on the default branch", async () => {
```

Comment:
You seem to validating calls to mocks in some tests and not in others. Why?

Reply:
The ones where I intentionally validate calls is b/c I cannot infer the lack of a call based on the return result of the function under test. I need to use `not.toHaveBeenCalled` in those cases

Reply:
so `mockResolvedValueOnce` will throw/cause errors if something has been called more than once, *and* if it wasn't called at all?

Reply:
no, im saying that i dont use any `not.toHaveBeenCalled` calls within the `versions` test block. I do see where I can remove a validation call now since we return an object that contains a test value rather than just a boolean. Making this change